src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,30 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
+        for (EventListener listener : listeners) {
+            executor.submit(() -> listener.onEvent(event));
+        }
 
     }
 
     public void shutdown() {
```

Comment:
When shutting down the executor, it's good practice to also await termination after shutdown to ensure that all tasks have completed execution. This avoids abrupt termination of running tasks.